src/site/content/en/blog/cache-api-quick-guide/index.md

```diff
@@ -0,0 +1,350 @@
+console.log(new Uint8Array(buffer));
+// Uint8Array(11) [72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100]
+```
+
+## Retrieving from a cache {: #retreiving }
+
+To find an item in a cache, you can use the `match` method.
+
+```js
+const response = await cache.match(request);
+console.log(request, response);
+```
+
+If `request` is a string it is first be converted to a `Request` by calling
+`new Request(request)`. The function returns a `Promise` that resolves to
+a `Response` if a matching entry is found, or `undefined` otherwise.
+
+To determine if two `Requests` match, more than just the URL is used. Two
+requests are considered different if they have different query strings,
+`Vary` headers and/or methods (`GET`, `POST`, `PUT`, etc.).
+
+You can ignore some or all of these things by passing an options object as a
+second parameter.
+
+```js
+const options = {
+  ignoreSearch: true,
+  ignoreMethod: true,
+  ignoreVary: true
+};
+
+const response = await cache.match(request, options);
+// do something with the response
+```
+
+If more than one cached request matches then the one that was created first is
+returned. If you want to retrieve *all* matching responses, you can use
+`cache.matchAll()`.
+
+```js
+const options = {
+  ignoreSearch: true,
+  ignoreMethod: true,
+  ignoreVary: true
+};
+
+const responses = await cache.matchAll(request, options);
+console.log(`There are ${responses.length} matching responses.`);
+```
+
+As a shortcut you can search over all caches at once by using `caches.match()`
+instead of calling `cache.match()` for each cache.
+
+## Searching
+
+The Cache API does not provide a way to search for requests or responses
+except for matching entries against a `Response` object. However, you can
+implement your own search using filtering or by creating an index.
+
+### Filtering
+
+One way to implement your own search is to iterate over all entries and
+filter down to the ones that you want. Let's say that you want to find all
+items that have URLs ending with '.png'.
+
+```js
+async function findImages() {
+  // Get a list of all of the caches for this origin
+  const cacheNames = await caches.keys();
+  const result = [];
+
+  for (const name of cacheNames) {
+    // Open the cache
+    const cache = await caches.open(name);
+
+    // Get a list of entries. Each item is a Request object
+    for (const request of await cache.keys()) {
+      // If the request URL matches, add the response to the result
+      if (request.url.endsWith('.png')) {
+        result.push(await cache.match(request));
+      }
+    }
+  }
+
+  return result;
+}
+```
+
+This way you can use any property of the `Request` and `Response` objects to
+filter the entries. Note that this is slow if you search over large sets of
+data.
+
+### Creating an index
+
+The other way to implement your own search is to maintain a separate index of
+entries that can be searched, stored in IndexedDB. Since this is the kind of
```

Comment:
```suggestion
entries that can be searched and store the index in IndexedDB. Since this is the kind of
```
Suggested rephrasing because I stumbled over this sentence and it took me a couple reads to figure out what it was saying